src/java/org/httpkit/client/HttpClient.java

```diff
@@ -454,13 +448,13 @@ private void processPending() {
                       ch.bind(bindAddress);
                     }
                     ch.configureBlocking(false);
+                    // save key for timeout check
+                    requests.offer(job);
                     boolean connected = ch.connect(job.addr);
                     job.setConnected(connected);
                     numConnections++;
                     // if connection is established immediatelly, should wait for write. Fix #98
                     job.key = ch.register(selector, connected ? OP_WRITE : OP_CONNECT, job);
-                    // save key for timeout check
```

Comment:
`job.setConnected()` may add to the `requests` queue, so we don't want to add a duplicate afterwards.